lib/child_process.js

```diff
@@ -183,24 +185,28 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
-    let resolve;
-    let reject;
-    const promise = new Promise((res, rej) => {
-      resolve = res;
-      reject = rej;
-    });
+  return (...file) => {
+    let child;
 
-    promise.child = orig(...args, (err, stdout, stderr) => {
-      if (err !== null) {
-        err.stdout = stdout;
-        err.stderr = stderr;
-        reject(err);
-      } else {
-        resolve({ stdout, stderr });
-      }
+    // Check if a callback is passed somehow
```

Comment:
Why this change?

Reply:
Just to check if somehow we're receiving a callback already. I think I missed the validation [here](https://github.com/nodejs/node/pull/34249/files/59cb6c1d6fa7c588d8127b95676a5e2a7bc54634#diff-56c0eb11e943939908cf3d1d251bf1aadda236b5adbf8111d79b0809ec403f97R242) in `execFile` fn. Let me remove it

Reply:
Actually, after doing a test, the Promise never gets resolve and the validation doesn't throw if a second callback is passed. Is this expected? Should something be thrown instead?

Reply:
I still don't understand why this change is related to this PR 😅

Reply:
Oh, got it! Sorry 🙏 
You're right might be out of the scope of the PR, let me revert it 😅